### PR TITLE
Add API methods to get schema information

### PR DESCRIPTION
**Pull request by pezholio**
This adds some API methods to return information about the schemas that Publishing API knows about, as well as fetching a schema by name. This allows us to get up to date schema information without having to have a local copy of the schemas stored somewhere and using the [GOV.UK Schemas](https://github.com/alphagov/govuk_schemas/) gem.

This will allow the content modelling team to have a way to create many different types of reusable content using the Publishing API as a single source of truth for the schema, rather than having to keep fields, validation rules etc in two places.

Happy to chat through in more detail if anyone needs any more context / has any questions!

### CODE REVIEW

File: spec/controllers/v2/schemas_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
+RSpec.describe V2::SchemasController do
```
Comment (Harriethw) [inline spec/controllers/v2/schemas_controller_spec.rb:1]: does there also need to be a request spec?

File: app/services/schema_service.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class SchemaService
+  def self.all_schemas
+    format_regex = Regexp.new(".+\/formats\/([a-z_]+)\/.+")
```
Comment (Harriethw) [inline app/services/schema_service.rb:3]: so here we are using the file path to get the type of object the schema is for? Is there any other way to get the type, e.g. in the schema itself with a `title` key for instance? Just wondering if there's a risk depending on a file path set in another gem which could easily change...

File: spec/controllers/v2/schemas_controller_spec.rb (hunk 1 above)
Comment (pezholio) [reply spec/controllers/v2/schemas_controller_spec.rb:1]: I guess I'm kinda doing a request spec over a controller spec here, as I guess controller specs are more designed to be "unit-type" tests, so I think I'll switch over 👍

File: app/services/schema_service.rb (hunk 2 above)
Comment (Harriethw) [reply app/services/schema_service.rb:3]: After chat with @pezholio There is no codified place for the name/type except in folder name

**Pull request merged**